XenAdmin/Wizards/NewNetworkWizard.cs

```diff
@@ -227,7 +227,17 @@ private void CreateSRIOV()
         private void CreateNonBonded()
         {
             XenAPI.Network network = PopulateNewNetworkObj();
-            PIF nic = pageNetworkDetails.SelectedHostNic;
+
+            PIF nic;
+            if(pageNetworkDetails.createVlanOnSriovNetwork)
+            {
+                var sriovLogicalPif = pageNetworkDetails.SelectedHostNic.sriov_physical_PIF_of[0];
+                nic = Pool != null ? Pool.Connection.Resolve(Pool.Connection.Resolve(sriovLogicalPif).logical_PIF) : Host.Connection.Resolve(Host.Connection.Resolve(sriovLogicalPif).logical_PIF);

```

Comment:
Can use xenConnection here. This won't be null.
Also to avoid nested Resolve calls here, we can set the sriovLogicalPif variable (should it be called sriovPhysicalPif?) to the resolved PIF object, on line 234.